upstairs/src/client.rs

```diff
@@ -380,16 +375,14 @@ impl DownstairsClient {
 
         // Update our bytes-in-flight counter
         if was_running && !is_running {
-            self.bytes_outstanding = self
-                .bytes_outstanding
-                .checked_sub(job.work.job_bytes())
-                .unwrap();
-            self.write_bytes_outstanding.decrement(job, self.client_id);
+            job.backpressure_guard.take(&self.client_id);
```

Comment:
I think I am understanding this correct: this takes the counter and drops it to do a decrement. And we do that here so that this job stops contributing to backpressure since it's not actually running. We expect this to be how most jobs decrement the backpressure counters, but if somehow they don't go through this path they will get cleaned up whenever that job finally gets dropped

Reply:
Yup, that's right.  I realized that my comment got lost; added it back in 7e11a7e3ccf5